### ISSUE TITLE
41303944: Users, SecureToken and MDM for non-DEP Macs

### ISSUE THREAD

**Issue by openradar-mirror** [lionheart/openradar-mirror#19997]
#### Description

Much of the frustration we have encountered with deploying macOS 10.13 High Sierra and expect to continue to have with macOS 10.14 Mojave (currently beta 2) boils down to issues with SecureToken. Specifically, SecureToken and how it is applied or not applied to users has wrecked havoc on our FileVault automations, which in turn impacts our deployments. 

We have almost 1800 Macs and practically none are DEP-enrolled. Over time the number of DEP macs will increase, but that will be a long process. Now and for the next couple of years/OS releases, we need a way to have non-DEP Macs have users we create through our management system (Jamf Pro) be granted SecureToken upon creation so that we can easily deploy FileVault.

Currently we use a policy action from Jamf Pro to create accounts on the machine. In 10.10 - 10.12, where SecureToken is not a factor, we have automations in place to properly add policy created users to FileVault. In 10.13 and 10.14, these accounts are not granted SecureToken due to the nature of how they were created. If we could create these accounts through an MDM command _and_ users created through MDM command were automatically granted SecureToken, we could once again automate FileVault deployment. Or if not an MDM command, if the system could somehow whitelist users based on a ConfigurationProfile and grant them SecureToken that would likely work too. 

tl;dr We need a way to automate putting a user on a 10.13+ Mac and have that user granted SecureToken.

-
Product Version: 10.13 & 10.14 Beta
Created: 2018-06-20T21:19:16.876499
Originated: 2018-06-20T00:00:00
Open Radar Link: http://www.openradar.me/41303944
